tests/parm/model_configure_hafs_shared.IN

```diff
@@ -25,19 +25,130 @@ kchunk3d:                -1
 ideflate:                @[IDEFLATE]
 nbits:                   @[NBITS]
 
-app_domain:              'regional'
-output_grid:             @[OUTPUT_GRID]
-cen_lon:                 @[CEN_LON]             # central longitude
-cen_lat:                 @[CEN_LAT]             # central latitude
-lon1:                    @[LON1]                # longitude of lower-left
-lat1:                    @[LAT1]                # latitude of lower-left
-lon2:                    @[LON2]                # longitude of upper-right
-lat2:                    @[LAT2]                # latitude of upper-right
-dlon:                    @[DLON]
-dlat:                    @[DLAT]
-
-nfhout:                  @[NFHOUT]
-nfhmax_hf:               @[NFHMAX_HF]
-nfhout_hf:               @[NFHOUT_HF]
-nsout:                   -1
-output_fh:               -1
+output_grid:             @[OUTPUT_GRID]     # gaussian_grid, global_latlon, regional_latlon, rotated_latlon, lambert_conformal
+imo:                     @[IMO]             # number of grid points along longitude/latitude direction
+jmo:                     @[JMO]             # needed for gaussian_grid or global_latlon
+cen_lon:                 @[CEN_LON]         # central longitude/latitude (degrees)
+cen_lat:                 @[CEN_LAT]         # needed for rotated_latlon, lambert_conformal
+lon1:                    @[LON1]            # longitude/latitude of lower-left corner
+lat1:                    @[LAT1]            # needed for regional_latlon, rotated_latlon, lambert_conformal
+lon2:                    @[LON2]            # longitude/latitude of upper-right corner
+lat2:                    @[LAT2]            # needed for regional_latlon, rotated_latlon
+dlon:                    @[DLON]            # grid spacing in longitude/latitude direction (degrees)
+dlat:                    @[DLAT]            # needed for regional_latlon, rotated_latlon
+stdlat1:                 @[STDLAT1]         # latitude of first/second standard parallel (degrees)
+stdlat2:                 @[STDLAT2]         # needed for lambert_conformal
+nx:                      @[NX]              # number of grid cells along x/y-axis
+ny:                      @[NY]              # needed for lambert_conformal
+dx:                      @[DX]              # grid cell size in x/y direction (meters)
+dy:                      @[DY]              # needed for lambert_conformal
+
+<output_grid_02>
```

Comment:
For the hafs test with single domain output, will this section still show up? Can we remove this section using hafs run script if these output grids are not needed?

Reply:
@junwang-noaa and @DusanJovic-NOAA, this is one of the reasons I suggested using the column configuration method. We do not need to add these extra <output_grid_xx> sections. And if default values are the concern, we can just use empty as default values (as what we are currently doing), saying.
```
output_grid:             'global_latlon', 'regional_latlon', 'rotated_latlon', 'rotated_latlon', 'rotated_latlon'
imo:                     360,   ,   ,   ,
jmo:                     181,   ,   ,   ,
cen_lon:                    , -175.00,  -86.00,  -86.00,  -86.00
cen_lat:                    ,    0.00,   23.00,   23.00,   23.00
lon1:                       , -220.00,  -45.00,  -45.00,  -35.00
lat1:                       ,  -45.00,  -45.00,  -45.00,  -30.00
lon2:                       , -130.00,   45.00,   45.00,   35.00
lat2:                       ,   45.00,   45.00,   45.00,   30.00
dlon:                       ,    0.25,    0.25,    0.10,    0.05
dlat:                       ,    0.25,    0.25,    0.10,    0.05
```
I personally think this will give us a much cleaner model_configure template/file. Thanks!

Reply:
> @junwang-noaa and @DusanJovic-NOAA, this is one of the reasons I suggested using the column configuration method. We do not need to add these extra <output_grid_xx> sections. And if default values are the concern, we can just use empty as default values, for example:
> 
> ```
> output_grid:             'global_latlon', 'regional_latlon', 'rotated_latlon', 'rotated_latlon', 'rotated_latlon'
> imo:                     360,   ,   ,   ,
> jmo:                     181,   ,   ,   ,
> cen_lon:                    , -175.00,  -86.00,  -86.00,  -86.00
> cen_lat:                    ,    0.00,   23.00,   23.00,   23.00
> lon1:                       , -220.00,  -45.00,  -45.00,  -35.00
> lat1:                       ,  -45.00,  -45.00,  -45.00,  -30.00
> lon2:                       , -130.00,   45.00,   45.00,   35.00
> lat2:                       ,   45.00,   45.00,   45.00,   30.00
> dlon:                       ,    0.25,    0.25,    0.10,    0.05
> dlat:                       ,    0.25,    0.25,    0.10,    0.05
> ```
> 
> I personally think this will give us a much cleaner model_configure template/file. Thanks!

Are you sure this table, with 'empty values', will properly parse using ESMF config parser? Shouldn't all values at a given row be the same type i.e. float?

Reply:
I am not very sure. I recall ESMF_config can treat every line as string, then one can parse it into variables (string, integer, float, etc.). But, @theurich probably know this better.

Reply:
You can certainly read in every line as a string. Of course, parsing this string, and tokenizing it then becomes your an application level responsibility.
If on the other hand you have as many elements in each line as there are columns, then you can use ESMF_Config API to read in each value. But that is where you would need dummies.